### PR TITLE
RD-261 Remove usages of deprecated React APIs

### PR DESCRIPTION
**Pull request by kubama**
## Description
Removed usages of deprecated React APIs

## Screenshots / Videos
N/A

## Checklist

- [X] My code follows the [UI Code Style](https://cloudifysource.atlassian.net/l/c/x8fq902p).
- [x] I verified that all tests and checks have passed.
- [X] I verified if that change requires any update in the [documentation](https://cloudifysource.atlassian.net/l/c/4XKtPocy).
- [X] I added proper labels to this PR.

## Tests
https://jenkins.cloudify.co/view/UI/job/Stage-UI-System-Test/3496/

 

## Documentation
N/A

### CODE REVIEW

File: app/components/Home.tsx [hunk 1]
```diff
@@ -20,25 +20,19 @@ import Page from './Page';
 import SideBar, { collapsedSidebarWidth, expandedSidebarWidth } from './sidebar/SideBar';
 
 class Home extends Component {
-    // TODO: Context handling should not be here. Currently necessary to use deprecated methods.
-    // eslint-disable-next-line camelcase
-    UNSAFE_componentWillMount() {
+    componentDidMount() {
```
Comment (qooban) [inline app/components/Home.tsx:23]: Are you sure drilldown context handling works properly after these changes?

I checked quickly and I'm seeing that:

https://user-images.githubusercontent.com/5202105/186853311-fc9fc635-5a46-463d-ab71-ddce629069d4.mp4

You can use `Nested-Example` blueprint from **Blueprint Marketplace > Getting Started** to reproduce it.

File: app/components/Home.tsx [hunk 2]
```diff
@@ -20,25 +20,19 @@ import Page from './Page';
 import SideBar, { collapsedSidebarWidth, expandedSidebarWidth } from './sidebar/SideBar';
```
Comment (qooban) [inline app/components/Home.tsx:20]: Maybe worth to migrate to TS at once?

**Review by qooban — COMMENTED** [review]
Added 2 comments.

Turned on Strict Mode in local environment and saw some warnings (about using `findDOMNode` and legacy context API). Are all of them coming from Semantic UI React and we can just wait for v3 or there are still places which we should fix on our side?

File: app/components/Home.tsx (hunk 2 above)
Comment (kubama) [reply app/components/Home.tsx:21]: Done

File: app/components/Home.tsx (hunk 1 above)
Comment (kubama) [reply app/components/Home.tsx:46]: I rechecked locally and it's working just fine for me. This is also covered by system tests which are passing. Could you please re-validate?

**Comment by kubama** [comment]
> Are all of them coming from Semantic UI React and we can just wait for v3 or there are still places which we should fix on our side?

I identified 2 more dependencies that have the same issue:
1. `react-visibility-sensor` - https://github.com/joshwnj/react-visibility-sensor/issues/141
The project seems to be dead, no fix can be expected.
2. `react-sortable-hoc` - https://github.com/clauderic/react-sortable-hoc/issues/677
Same as above.

I think we should have separate subtasks for those, however as fixes involve replacing these with actively maintained equivalents then these subtasks can be quite time consuming, so I'm not sure if I should include them or they should be planned separately.

File: app/components/Home.tsx (hunk 1 above)
Comment (qooban) [reply app/components/Home.tsx:46]: OK, I'll recheck and let you know.

Comment (qooban) [reply app/components/Home.tsx:46]: Rechecked and it works fine. Sorry for misleading. There was probably an issue in my local environment.

**Review by qooban — APPROVED** [review]
Thanks for additional effort on TS migration. LGTM 👍🏻 

> I identified 2 more dependencies that have the same issue:
> 
> 1. `react-visibility-sensor` - [findDOMNode will be deprecated joshwnj/react-visibility-sensor#141](https://github.com/joshwnj/react-visibility-sensor/issues/141)
>    The project seems to be dead, no fix can be expected.
> 2. `react-sortable-hoc` - [React's StrictMode Warnings clauderic/react-sortable-hoc#677](https://github.com/clauderic/react-sortable-hoc/issues/677)
>    Same as above.
> 
> I think we should have separate subtasks for those, however as fixes involve replacing these with actively maintained equivalents then these subtasks can be quite time consuming, so I'm not sure if I should include them or they should be planned separately.

Yes, we should have seperate subtasks for them.

What do you think about creating similar ticket to RD-261 containing 3 subtasks - 1. RD-263, 2. for `react-visibility-sensor` replacement, 3. for `react-sortable-hoc` replacement? If that's fine, could you create it?

That will allow us to close RD-261 with what we have and continue this activity in the future (not in this sprint) with a new ticket.

**Review by Vorbert-Kruk — APPROVED** [review]
LGTM 🎉

**Comment by kubama** [comment]
> What do you think about creating similar ticket to [RD-261](https://cloudifysource.atlassian.net/browse/RD-261) containing 3 subtasks - 1. [RD-263](https://cloudifysource.atlassian.net/browse/RD-263), 2. for `react-visibility-sensor` replacement, 3. for `react-sortable-hoc` replacement? If that's fine, could you create it?

Created RD-5688.

**Pull request merged**